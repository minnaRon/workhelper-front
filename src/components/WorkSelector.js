/**
 * Module, component Form WorkSelector
 * @description User selects current work using this form.
 * The user selects the name of the work, the way of working
 * * and marks the work as a project when it is a project.
 * New work is possible to add using input fields;
 * * newWorkName, newWorkType and checking project checkbox as needed.
 * exports WorkSelector as default
 */
import { useState, useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useNavigate } from 'react-router-dom'
import { addWork } from '../reducers/worksRed'
import { updateUser } from '../reducers/usersRed'
import { updateWork } from '../reducers/worksRed'
import TextField from '@mui/material/TextField'
import Autocomplete from '@mui/material/Autocomplete'
import FormControlLabel from '@mui/material/FormControlLabel'
import Checkbox from '@mui/material/Checkbox'
import Typography from '@mui/material/Typography'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'

const WorkSelector = () => {
  const works = useSelector((state) => state.works)
  const user = useSelector((state) => state.user)
  const vocabulary = useSelector((state) => state.vocabulary)
  const v = vocabulary.vocabulary.checked.workselector

  const [selectedWork, setSelectedWork] = useState(works[0] || 'LISÄÄ ENSIMMÄINEN TYÖ')
  const [isProject, setIsProject] = useState(selectedWork.isProject || false)
  const [workType, setWorkType] = useState(selectedWork.type || 'working at home')
  const [newWorkName, setNewWorkName] = useState('')
  const [newWorkType, setNewWorkType] = useState('')
  const dispatch = useDispatch()
  const navigate = useNavigate()

  if (!user) {
    navigate('/')
  }

  /**
   * hook useEffect
   * Executed if the state of the works or the user changes.
   * @description Sets default selected options using first work object in the array of the works
   * which is sorted by lastVisited work object to be first in the array.
   */
  useEffect(() => {
    if (works[0]) {
      setSelectedWork(works[0])
      setIsProject(selectedWork.isProject)
      setWorkType(selectedWork.type)
    }
  }, [works, user])

  /**
   * function isNew
   * @description Checks if user's newWorkType input already exists
   * in the array of the user's work types.
   * @returns true if work type is new, false if already exists.
   * Is case sensitive; 'Alpha' and 'alpha' not the same.
   * */
  const isNew = (newWorkType) => {
    const workTypes = user.workTypes
    return workTypes.find((type) => type === newWorkType) === undefined
  }

  /**
   * @description Function handles submit of the information of the selected work.
   * @param {*} event - not in use here, uses state values.
   * Checks if type of working is new and typed in the input field newWorkType.
   * * if type of working is new, dispatch update to user's workTypes.
   * Checks if work is new and typed in the input field newWorkName.
   * * if is new work, dispatch to add it in the database and user's info.
   * If work selected using already existing options,
   * * dispatches update using selected options to the information of the existing work.
   */
  const handleSubmit = (event) => {
    event.preventDefault()
    if (newWorkType && isNew(newWorkType)) {
      const updatedWorkTypes = [...user.workTypes, newWorkType]
      dispatch(updateUser({ workTypes: updatedWorkTypes }))
    }
    if (newWorkName) {
      const newWork = {
        name: newWorkName,
        isProject: isProject || false,
        type: newWorkType || workType,
      }
      dispatch(addWork(newWork))
    } else {
      dispatch(updateWork({ ...selectedWork, isProject, type: newWorkType || workType }))
    }
    setNewWorkName('')
    setNewWorkType('')
  }

  /**
   * list optionsSelectWork
   * @description option list for work select menu
   * @returns list of the select options of the work names to use in the select menu.
   */
  const optionsSelectWorkForAutoComplete = works.map((work) => ({
    ...work,
    label: work.name,
  }))

  /**
   * handler changeWorkHandler
   * @description handles the work selection made by the user.
   * Sets correct values of the selected work for local states: selectedWork, isProject, workType
   * These values will be shown in the form as default before submit is done.
   */
  const workHandler = (work) => {
    if (work) {
      setSelectedWork(work)
      setIsProject(work.isProject)
      setWorkType(work.type)
    }
  }

  /**
   * list optionsRadioWorkType
   * @description radio option list of types of working.
   * @returns list of the radio options of the work types.
   */
  const optionsSelectWorkTypeForAutoComplete = user.workTypes.map((type) => ({
    type,
    label: type,
  }))

  return (
    <form onSubmit={handleSubmit}>
      <Grid container pb={4} align="start" justifyContent="center" rowSpacing={2}>
        <Grid item xs={12} mr={7}>
          <Typography variant="h5">{v.WSH1headlineT}</Typography>
        </Grid>
        <Grid item xs={12}>
          <Autocomplete
            disabled={newWorkName.length > 0}
            isOptionEqualToValue={(option, value) => option.id === value.id}
            disablePortal
            size="small"
            id="workSelector-select-work"
            value={selectedWork.name}
            options={optionsSelectWorkForAutoComplete}
            sx={{ width: 300 }}
            renderInput={(params) => <TextField {...params} label={v.WSSworkT} />}
            onChange={(event, newValue) => {
              workHandler(newValue)
            }}
          />
        </Grid>
        <Grid item xs={12}>
          <TextField
            size="small"
            label={v.WSInewWorkT}
            id="workSelector-input-newWork"
            type="text"
            value={newWorkName}
            placeholder={v.WSInewWorkP}
            onChange={({ target }) => {
              setNewWorkName(target.value)
            }}
          />
        </Grid>
        <Grid item xs={12}>
          <FormControlLabel
            id="workSelector-checkbox-isProject"
            control={<Checkbox checked={isProject || false} />}
            label={v.WSCisProjectT}
            onChange={(e) => setIsProject(e.target.checked)}
          />
        </Grid>
        <Grid item xs={12}>
          <Autocomplete
            disabled={newWorkType.length > 0}
            isOptionEqualToValue={(option, value) => option.type === value}
            disablePortal
            value={workType || ''}
            size="small"
            id="workSelector-select-newWorkTypek"
            options={optionsSelectWorkTypeForAutoComplete}
            sx={{ width: 300 }}
            renderInput={(params) => <TextField {...params} label={v.WSRworkTypeT} />}
            onChange={(event, newValue) => {
              setWorkType(newValue.type)
            }}
          />
        </Grid>
        <Grid item xs={12}>
          <TextField
            size="small"
            label={v.WSInewWorkTypeT}
            id="workSelector-input-newWorkType"
            type="text"
            value={newWorkType}
            placeholder={v.WSInewWorkTypeP}
            onChange={({ target }) => {
              setNewWorkType(target.value)
            }}
          />
        </Grid>
        <Grid item xs>
          <Button variant="contained" type="submit" id="workSelector-button-submit">
            {v.WSBsubmitT}
          </Button>
        </Grid>
      </Grid>
    </form>
  )
}

export default WorkSelector
