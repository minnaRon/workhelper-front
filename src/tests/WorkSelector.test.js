/**
 * snapshot
 */
import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import '@testing-library/jest-dom/extend-expect'
import WorkSelector from '../components/WorkSelector'

const mockStore = configureStore([])

describe('<WorkSelector />', () => {
  let store
  let component

  beforeEach(() => {
    store = mockStore({
      user: {
        username: 'testUsername',
        name: 'testName',
        works: [{ id: 'testWork1Id' }, { id: 'testWork2Id' }],
        workTypes: ['working at home', 'studying'],
        password: 'testPassword',
      },
      works: [
        {
          name: 'testWork1',
          isProject: true,
          type: 'working at home',
          active: true,
          lastWorked: '2023-04-17T01:42:12.746Z',
          id: 'testWork1Id',
        },
        {
          name: 'testWork2',
          isProject: false,
          type: 'studying',
          active: true,
          lastWorked: '2023-04-17T01:42:12.746Z',
          id: 'testWork2Id',
        },
      ],
      vocabulary: {
        vocabulary: {
          checked: {
            workselector: {
              WSH1headlineT: 'Työn iloihin!',
              WSSworkT: 'TYÖN KOHDE: ',
              WSInewWorkT: '...TAI LISÄÄ UUSI: ',
              WSInewWorkP: 'LISÄÄ UUSI TYÖ',
              WSCisProjectT: 'ON PROJEKTI',
              WSRworkTypeT: 'TYÖTAPA: ',
              WSInewWorkTypeT: '...TAI LISÄÄ UUSI: ',
              WSInewWorkTypeP: 'LISÄÄ UUSI TYÖTAPA',
              WSBsubmitT: 'ALOITA TYÖT',
            },
          },
        },
      },
    })

    component = renderer.create(
      <Provider store={store}>
        <Router>
          <WorkSelector />
        </Router>
      </Provider>
    )
  })

  test('renders correct empty form (this fails also if local vocabulary changes', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
