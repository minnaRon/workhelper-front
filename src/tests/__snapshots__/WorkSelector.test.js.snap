// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkSelector /> renders correct empty form (this fails also if local vocabulary changes 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    align="start"
    className="MuiGrid-root MuiGrid-container css-gj9ng7-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49awxh-MuiGrid-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Työn iloihin!
      </h5>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        aria-owns={null}
        className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-1fvqjm9-MuiAutocomplete-root"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="workSelector-select-work"
            id="workSelector-select-work-label"
          >
            TYÖN KOHDE: 
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="workSelector-select-work"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              readOnly={false}
              required={false}
              role="combobox"
              spellCheck="false"
              type="text"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Clear"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </button>
              <button
                aria-label="Open"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  TYÖN KOHDE: 
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="workSelector-input-newWork"
          id="workSelector-input-newWork-label"
        >
          ...TAI LISÄÄ UUSI: 
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="workSelector-input-newWork"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="LISÄÄ UUSI TYÖ"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                ...TAI LISÄÄ UUSI: 
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
        id="workSelector-checkbox-isProject"
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          ON PROJEKTI
        </span>
      </label>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        aria-owns={null}
        className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-1fvqjm9-MuiAutocomplete-root"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="workSelector-select-newWorkTypek"
            id="workSelector-select-newWorkTypek-label"
          >
            TYÖTAPA: 
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="workSelector-select-newWorkTypek"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              readOnly={false}
              required={false}
              role="combobox"
              spellCheck="false"
              type="text"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Clear"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </button>
              <button
                aria-label="Open"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  TYÖTAPA: 
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="workSelector-input-newWorkType"
          id="workSelector-input-newWorkType-label"
        >
          ...TAI LISÄÄ UUSI: 
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="workSelector-input-newWorkType"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="LISÄÄ UUSI TYÖTAPA"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                ...TAI LISÄÄ UUSI: 
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        id="workSelector-button-submit"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        ALOITA TYÖT
      </button>
    </div>
  </div>
</form>
`;
